### DISCUSSION TITLE
nwProject.lock

### DISCUSSION THREAD

**Discussion by relogi** [vkbo/novelWriter#694] [Help] [answered]
is there a way to disable this feature? I work on a couple of different computers during a typical week (sharing an office with my wife, there are times I cant work there) and I end up deleting .lock files and conflicted Project files just to access a Project. I am worried about the stability of novelWriter with the way I treat it. Please advise

**Comment by vkbo** [comment, marked as answer]
It's a safety feature to prevent you from overwriting your own data, but you can click yes to ignore the warning and continue, which should force novelWriter to proceed as normal. Ideally, you should close the project on one computer before opening on another. In the end, the warning is just a warning, and the lock file only exists to provide that warning. It is there for a reason, but selecting to override it has no other consequences for the functionality of the application.

The only real risk of data loss is if you have the same text file within a project open in both places and have made different changes to it on each computer. The two versions of the file will just overwrite each other and the last one to be closed will get the last word. If you have file sync software, it may register the conflict and mark the files as conflicting in such cases. These are the files in the `contents` folder in your project disk location. All text you write ends up there, and novelWriter doesn't even touch that data unless a file is open in the editor window.

The lesser risk is that if you add a new file to the project on one end, the other one will not know about it until the next time the project is opened (it scans everything when you open a project). It may then find "orphaned files", which will lack some of the settings you may have picked for it. However, the content of the file will be fine. Files found this way will be re-introduced into the project, and marked so you know it has been recovered. novelWriter will do its best to put it in the correct place, but you may have to make the final decisions. This is a case of diverging `nwProject.nwx` files. This is the XML file that describes your project. You can have a look at it, it's fairly readable.

A non-critical risk is that your project index may also drift out of sync, which means your word counts may be wrong and tags made on one computer may not be known on the other. But all of that can be fixed by pressing `F9` to rebuild the index, so that is more of an inconvenience than a problem.

At the end of the day, novelWriter works with human readable text files for a reason. It makes it less vulnerable to project-wide file conflicts and easier for file sync software to deal with. I am considering adding a few more safety features to prevent overwriting conflicting files like may occur in this type of cases, but it will not resolve the issue with diverging project files.

**Reply by relogi** [reply]
i must be doing something wrong. I not only close out of novelWriter, but the computer in question is shut down when I log onto the other computer and open up novelWriter on it. At no time am I on one without the other being shutdown. Is it possible that a pCloud storage instance is considered open? Even if novelWriter is closed before shutting computer down? 
Thank you for your response, btw, i feel better at letting the program yell at me...

**Reply by vkbo** [reply]
Ah, I thought you were just moving between two running computers to pick up your work where you left off.

I think I know what's going on then. I've encountered this problem in an entirely different application. If you close an application, which does a bit of its own cleanup, and then suspend or shut down that computer before the file sync software is done syncing, it may not manage to finish and leave the folder in an intermediate state. novelWriter deletes the lock file when you close the project. It is the very last thing it does after saving everything else.

Could it simply be incomplete file synchronisation? If so, I just propose to give it a bit more time, or force a synchronisation if possible. I'm not familiar with pCloud, but I've had this issue with both Dropbox and Syncthing.

This is getting me thinking I should add a way to validate the content of the project to make sure it is in a consistent state.

**Reply by vkbo** [reply]
> Thank you for your response, btw, i feel better at letting the program yell at me...

I'm a bit paranoid about losing my work after I lost a week of work back in 2004, so I suppose that is somewhat reflected in the way I wrote novelWriter!

**Reply by relogi** [reply]
love your paranoia (you rock)and I bet it is I'm closing up too quick because some times it doesnt happen. excellent.

**Reply by vkbo** [reply]
Thanks!
